native/src/components/LanguageNotAvailablePage.tsx

```diff
@@ -5,37 +5,54 @@ import styled from 'styled-components/native'
 
 import { LanguageModel } from 'api-client'
 
+import useCityAppContext from '../hooks/useCityAppContext'
+import useLoadLanguages from '../hooks/useLoadLanguages'
 import SelectorItemModel from '../models/SelectorItemModel'
+import LoadingErrorHandler from '../routes/LoadingErrorHandler'
 import Caption from './Caption'
 import Selector from './Selector'
 
 const Wrapper = styled.ScrollView`
   background-color: ${props => props.theme.colors.backgroundColor};
 `
 
-export type LanguageNotAvailablePageProps = {
-  languages: Array<LanguageModel>
-  changeLanguage: (newLanguage: string) => void
+type LanguageNotAvailablePageProps = {
+  languages?: LanguageModel[]
+  changeLanguage?: (newLanguage: string) => void
 }
 
 const LanguageNotAvailablePage = ({ languages, changeLanguage }: LanguageNotAvailablePageProps): ReactElement => {
+  const { cityCode, changeLanguageCode } = useCityAppContext()
+  const { data, ...response } = useLoadLanguages({ cityCode })
   const { t } = useTranslation('common')
-  const selectorItems = languages.map(
+
+  const availableLanguages = languages ?? data
+
+  const items = availableLanguages?.map(
     ({ code, name }) =>
       new SelectorItemModel({
         code,
         name,
         enabled: true,
-        onPress: () => changeLanguage(code),
+        onPress: () => {
+          changeLanguageCode(code)
+          if (changeLanguage) {
+            changeLanguage(code)
```

Comment:
Why do we need both: the changeLanguageCode-function and the changeLanguage-function?

Reply:
One is legacy for routes that still use the redux store and will be removed later once the redux store gets kicked out completely.